docs/high/dataset.rst

```diff
@@ -614,3 +614,16 @@ Reference
     .. attribute:: parent
 
         :class:`Group` instance containing this dataset.
+
+    .. attribute:: __bool__
+
+        Check that the dataset is accessible:
```

Comment:
I think we should say that you can use this to check if the file it belongs to has been closed. It's clear in the example, but a bit of repetition doesn't hurt to make things clear.

There are other ways you can make an object invalid, like calling `ds.id.close()`, so it's not only telling you if the file is still open, but I think that's the main useful scenario.

Reply:
That makes sense, I've clarified that datasets can be inaccessible for various reasons.